src/pal/src/loader/module.cpp

```diff
@@ -1295,88 +1430,82 @@ static MODSTRUCT *LOADAllocModule(void *dl_handle, LPCSTR name)
     return module;
 }
 
-/*++
-Function :
-    LOADLoadLibrary [internal]
-
-    implementation of LoadLibrary (for use by the A/W variants)
-
-Parameters :
-    LPSTR shortAsciiName : name of module as specified to LoadLibrary
+/*
+Function:
+    LOADLoadLibraryDirect [internal]
 
-    BOOL fDynamic : TRUE if dynamic load through LoadLibrary, FALSE if static load through RegisterLibrary
+    Loads a library using a system call, without registering the library with the module list.
 
-Return value :
-    handle to loaded module
+Parameters:
+    LPCSTR libraryNameOrPath:           The library to load.
+    bool setLastError:                  True to set the last error upon failure, false otherwise.
+    bool isModuleListLockedByCaller:    True if the caller already owns the module list lock, false otherwise.
 
---*/
-static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
+Return value:
+    System handle to the loaded library, or nullptr upon failure (error is set via SetLastError() if 'setLastError' is true).
+*/
+static HMODULE LOADLoadLibraryDirect(LPCSTR libraryNameOrPath, bool setLastError, bool isModuleListLockedByCaller)
 {
-    CHAR fullLibraryName[MAX_LONGPATH];
-    MODSTRUCT *module = NULL;
-    void *dl_handle = NULL;
-    DWORD dwError;
-    DWORD retval;
+    _ASSERTE(libraryNameOrPath != nullptr);
+    _ASSERTE(libraryNameOrPath[0] != '\0');
 
-    // Check whether we have been requested to load 'libc'. If that's the case then use the
-    // full name of the library that is defined in <gnu/lib-names.h> by the LIBC_SO constant.
-    // The problem is that calling dlopen("libc.so") will fail for libc even thought it works
-    // for other libraries. The reason is that libc.so is just linker script (i.e. a test file).
-    // As a result, we have to use the full name (i.e. lib.so.6) that is defined by LIBC_SO.
-    if (strcmp(shortAsciiName, LIBC_NAME_WITHOUT_EXTENSION) == 0)
+    if (!isModuleListLockedByCaller)
     {
-#if defined(__APPLE__)
-        shortAsciiName = "libc.dylib";
-#elif defined(__FreeBSD__)
-        shortAsciiName = FREEBSD_LIBC;
-#else
-        shortAsciiName = LIBC_SO;
-#endif
+        LockModuleList();
     }
 
-    LockModuleList();
-
-    // See if file can be dlopen()ed; this should work even if it's already loaded
+    HMODULE dl_handle = dlopen(libraryNameOrPath, RTLD_LAZY | RTLD_NOLOAD);
+    if (dl_handle == nullptr)
```

Comment:
I didn't see a reason for it, thought it was a workaround for something. I'll remove it.